### PR TITLE
Image index

### LINKED ISSUE #8: Image Index
#### As a user I want the homepage to display all saved images.

__Story__:

Now that we are saving images for our users, we want them to be able to view
the list of images in the system. You might be thinking, "what if my users only
want to view images of cats", or "what if my users only want to see the images
they uploaded". Those are great questions. Maybe you will have the opportunity
to add those features in a future story. For now, we will keep it simple and
quickly deliver a little more value to our users.

__Acceptance criteria__:
- [ ] New images that are added show up on the homepage.
- [ ] These images are persisted if the browser is closed or even if the
  server is restarted.
- [ ] Images are not displayed wider than 400px.
- [ ] Newest images appear first.

__Dependencies__:
- Save Image Link

### PR DESCRIPTION
**Pull request by NilsBacke**
Closes #8 

---

The following is a checklist to remind you of the steps to take when opening a
new PR. The horizontal line above and everything below can be removed from the
pull request description.

- [ ] I have replaced `X` above with the associated issue number.
- [ ] I have changed the issue's label to "pull request".
- [ ] I have assigned the designated reviewer to this PR.
- [ ] I have set the appropriate label on this PR.
- [ ] I have written clear commit messages ([ref](http://chris.beams.io/posts/git-commit/)).

If needed, please review the document on the
[Image Sharing Project](https://sites.google.com/a/appfolio.com/eng/new-hire-resources/engineering-academy/ropes/image-sharing-project)
for information on how labels should be used.

### CODE REVIEW

**Comment by NilsBacke** [comment]
[Image index (home page) - Nils](https://trello.com/c/cH2iBoxk/643-image-index-home-page-nils)

File: app/views/images/index.html.erb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+<h1>All Images</h1>
+<%= link_to 'Add new image', new_image_path %><br>
```
Comment (joelbandi) [inline app/views/images/index.html.erb]: Now might be a good time to introduce bootstrap for rails

Comment (joelbandi) [reply app/views/images/index.html.erb]: We can wrap these in a table or a card component from bootstrap

File: test/controllers/images_controller_test.rb [hunk 2]
```diff
@@ -22,4 +22,46 @@ def test_assigns_image_in_show
     assert_equal assigns(:image).id, image.id
     image.destroy!
   end
+
+  def test_assigns_images_in_index
+    Image.destroy_all
```
Comment (joelbandi) [inline test/controllers/images_controller_test.rb]: Not sure as to why we are destroying the images.

File: test/controllers/images_controller_test.rb [hunk 3]
```diff
@@ -22,4 +22,46 @@ def test_assigns_image_in_show
...
+    assert_response :ok
+
+    assigns(:images).each_cons(2) do |imgpair|
+      assert imgpair[0].created_at >= imgpair[1].created_at
+    end
+  end
+
+  private
+
+  def create_images
```
Comment (joelbandi) [inline test/controllers/images_controller_test.rb]: We can move this to `setup`

File: app/views/images/index.html.erb (hunk 1 above)
Comment (NilsBacke) [reply app/views/images/index.html.erb]: I'll add this in the image tags PR!

Comment (joelbandi) [reply app/views/images/index.html.erb]: Sounds good 👍

File: test/controllers/images_controller_test.rb (hunk 2 above)
Comment (NilsBacke) [reply test/controllers/images_controller_test.rb]: In order for me to assert_empty on `@images`, I must make sure that the image table is empty. The image table gets written to in `test_assigns_nonempty_images_in_index` and `test_images_are_sorted`

**Pull request merged**